doc/todo/RELEASE

```diff
@@ -9,6 +9,59 @@ Following shell expansions are used in this document:
  ~elektra-ftp .. where ftp files go to
  ~elektra-doc .. where doc flies go to
 
+----------------------------------------------------------------------
+
+This describes what to do for the automated release:
+
+= Tasks before Release =
+
+== Updates ==
+
+- Update doc/COMPILE.md to reflect actually tested setups
+- Write release notes
+
+== Increment Version Number ==
+
+- CMakeLists.txt
+- ~e/scripts/dev/build-debian-package
+- Change VERSION variable in build-server
+- Commit and push the changes
+
+== Check ==
+
+- Everything blue on https://build.libelektra.org/
+- Trigger release pipeline with dry run
+- Inspect the logs, artifacts and the website from the pipeline
+  - [ ] error logs (*.error) must be empty
+  - [ ] check in the (test-*) folder if all tests passed
+  - [ ] check if version is correct in the version file
+  - [ ] 
```

Comment:
inspect the changes done to the git repo, like what scripts/dev/update-infos-status did.

Maybe you can also add a `git diff` and `git log -p` of what happened in the release pipeline.

Reply:
Regarding the `git diff` and `git log -p`, do you think it would be better to 
1. run these commands inside the release pipeline and add the output to the artifacts or
2. keep archiving the git repositories in the dry-run option and run these commands locally?

I would prefer former since the latter option would introduce an overhead.

Reply:
I agree we should avoiding archiving the whole git repo. And storing the whole diff is probably also too big and not really needed. Actually, we only need to create the git statistics in the pipeline and add the output to the artifacts.

Copying the statistics into the release notes can be a manual step, as usually you need to modify them a bit (formatting, remove uninteresting data and so on...). As this is always a bit subjective, it is actually good if the raw statistics are somewhere else (in the artifacts), too.

Reply:
More convenient probably would be if you actually create the release notes (replace the VERSION, copy the statistics and hash sums there and so on). After this is done, the release author simply modifies (e.g. reformat) what was published from the release pipeline. This would, however, already be a quite advanced convenience-feature and is definitely not a must-have.

Reply:
I agree, archiving the git repos was only an intermediate step.

Do you think it makes sense to use a manual step in the pipeline to approve the changes before pushing evreything? (e.g. https://www.jenkins.io/doc/pipeline/steps/pipeline-input-step/)
The goal would be to prevent that an accidental error in the release pipeline is pushed to master. On the other hand it's not very elegant when everything goes right anyway.

Reply:
> Do you think it makes sense to use a manual step in the pipeline to approve the changes before pushing evreything? (e.g. https://www.jenkins.io/doc/pipeline/steps/pipeline-input-step/)

I think the manual step would be a good alternative to the dry-run option. We could inspect the artifacts, git statistics etc. without running the release pipeline twice. This would also prevent pushing errors to master if something went wrong during the non-dry-run.

Reply:
Yes, I also like the idea of the manual input instead of the dry-run. I didn't know it is possible. I think we can give it a try.